src/validation.cpp

```diff
@@ -4303,7 +4304,15 @@ bool LoadMempool(void)
                 if (state.IsValid()) {
                     ++count;
                 } else {
-                    ++failed;
+                    // mempool may contain the transaction already, e.g. from
+                    // wallet(s) having loaded it while we were processing
+                    // mempool transactions; consider these as valid, instead of
+                    // failed, but mark them as 'already existed'
+                    if (mempool.exists(tx->GetHash())) {
```

Comment:
Yes. There's a race condition going on here, which makes this better than the alternative (i.e. checking if exists and then trying to accept).

Thread A is doing wallet mempool insertions.
Thread B is calling LoadMempool().

Case 1:
- Thread A imports tx X
- Thread B tests for presence of tx X. Found. Marks already there and moves on.

Case 2:
- Thread B tests for presence of tx X. Not found.
- Thread A imports tx X.
- Thread B attempts to import tx X and fails.

Both cases should result in 'already there', but by checking presence before attempt to accept, the second case will result in a failure.